### ISSUE TITLE
diff headers give error

### ISSUE THREAD

**Issue by piranna** [kpdecker/jsdiff#76]
When patching with https://github.com/maximeh/buildroot/blob/0400322f2d4e28faa98e31815921f54106aeb2e6/package/openssl/openssl-004-musl-termios.patch, it give the next error:

```javascript
/home/piranna/Proyectos/NodeOS/node_modules/diff/lib/patch/parse.js:44
        throw new Error('Unknown line ' + (i + 1) + ' ' + JSON.stringify(diffs
              ^
Error: Unknown line 1 "http://rt.openssl.org/Ticket/Display.html?id=3123"
    at parseIndex (/home/piranna/Proyectos/NodeOS/node_modules/diff/lib/patch/parse.js:44:15)
    at Object.parsePatch (/home/piranna/Proyectos/NodeOS/node_modules/diff/lib/patch/parse.js:120:5)
    at applyPatch (/home/piranna/Proyectos/NodeOS/node_modules/diff/lib/patch/apply.js:13:22)
    at /home/piranna/Proyectos/NodeOS/node_modules/nodeos-barebones/scripts/preinstall:56:30
    at fs.js:334:14
    at FSReqWrap.oncomplete (fs.js:95:15)
```

Seems that after adding support for [multiple indexes](https://github.com/kpdecker/jsdiff/commit/532bcf9315a39518a772e47d912fb0d22b1aa701) this got broken and now the first comment lines of the patch don't get ignored.

**Comment by piranna** [comment]
I've checked it and the error was introduced on version 2.0.0

**Comment by kpdecker** [comment]
I've relaxed the parser to ignore unexpected lines unless in strict mode.

**Comment by kpdecker** [comment]
Released in 2.1.2

**Comment by piranna** [comment]
Cool, thank you! :-)
